extensions/pkg/controller/healthcheck/worker/nodes.go

```diff
@@ -162,7 +162,7 @@ func (h *DefaultHealthChecker) Check(ctx context.Context, request types.Namespac
 			return &healthcheck.SingleCheckResult{
 				Status: gardencorev1beta1.ConditionFalse,
 				Detail: err.Error(),
-				Codes:  gardencorev1beta1helper.DeprecatedDetermineErrorCodes(err),
+				Codes:  gardencorev1beta1helper.DeprecatedDetermineErrorCodes(err), // TODO(@acumino): Drop error detection once extensions get adapted to parse error code from `SingleCheckResult`.
```

Comment:
Can you adapt this 
https://github.com/gardener/gardener/blob/cffdb9fd10d598be614d90055ef1bf79d4e5a791/extensions/pkg/controller/healthcheck/actuator.go#L64-L68
struct to register some function which can take in the error and return errorcodes,  so that the extensions, https://github.com/gardener/gardener-extension-provider-aws/blob/11bdd4996cda5c90ae1162748466e5a0d73f54b9/pkg/controller/healthcheck/add.go#L75
can have their own error code detection functions (first with the legacy regex and then they can imrpove it in the future).

Reply:
PTAL a0440e263cc9d243e8b6cafc0684134256745339